qcodes/instrument_drivers/Keysight/Keysight_81180A.py

```diff
@@ -53,360 +76,399 @@ def __init__(self, parent, name, id, **kwargs):
+            "\ttime: Perform trigger at fixed time interval.\n"
+            "\tdelay: Perform trigger at fixed time intervals after "
+            "previous waveform finished.",
+        )
+
+        self.add_parameter(
+            "trigger_timer_time",
+            get_cmd="TRIGGER:TIMER:TIME?",
+            set_cmd="TRIGGER:TIMER:TIME {:f}",
             get_parser=float,
-            unit='s',
+            unit="s",
             vals=vals.Numbers(100e-9, 20),
-            docstring='Triggering period when in timer trigger mode')
+            docstring="Triggering period when in timer trigger mode",
+        )
 
         self.add_parameter(
-            'trigger_timer_delay',
-            get_cmd='TRIGGER:TIMER:DELAY?',
-            set_cmd='TRIGGER:TIMER:DELAY {:d}',
+            "trigger_timer_delay",
+            get_cmd="TRIGGER:TIMER:DELAY?",
+            set_cmd="TRIGGER:TIMER:DELAY {:d}",
             get_parser=int,
-            unit='1/sample rate',
-            vals=vals.Multiples(min_value=152, max_value=8000000, divisor=8))
+            unit="1/sample rate",
+            vals=vals.Multiples(min_value=152, max_value=8000000, divisor=8),
+        )
 
         self.add_parameter(
-            'trigger_level',
-            get_cmd='TRIGGER:LEVEL?',
-            set_cmd='TRIGGER:LEVEL {:.4f}',
+            "trigger_level",
+            get_cmd="TRIGGER:LEVEL?",
+            set_cmd="TRIGGER:LEVEL {:.4f}",
             get_parser=float,
             vals=vals.Numbers(-5, 5),
-            unit='V',
-            docstring='Trigger level when trigger_mode is TTL')
+            unit="V",
+            docstring="Trigger level when trigger_mode is TTL",
+        )
 
         self.add_parameter(
-            'trigger_slope',
-            get_cmd='TRIGGER:SLOPE?',
-            set_cmd='TRIGGER:SLOPE {}',
-            vals=EnumVisa('POSitive', 'NEGative', 'EITher'))
+            "trigger_slope",
+            get_cmd="TRIGGER:SLOPE?",
+            set_cmd="TRIGGER:SLOPE {}",
+            vals=EnumCaseInsensitive("POSitive", "NEGative", "EITher"),
+        )
 
         self.add_parameter(
-            'trigger_delay',
-            get_cmd='TRIGGER:DELAY?',
-            set_cmd='TRIGGER:DELAY {:.4f}',
+            "trigger_delay",
+            get_cmd="TRIGGER:DELAY?",
+            set_cmd="TRIGGER:DELAY {:.4f}",
             get_parser=float,
-            unit='1/sample rate',
-            vals=vals.Multiples(min_value=0, max_value=8000000, divisor=8))
+            unit="1/sample rate",
+            vals=vals.Multiples(min_value=0, max_value=8000000, divisor=8),
+        )
 
         self.add_parameter(
-            'burst_count',
-            get_cmd='TRIGGER:COUNT?',
-            set_cmd='TRIGGER:COUNT: {}',
+            "burst_count",
+            get_cmd="TRIGGER:COUNT?",
+            set_cmd="TRIGGER:COUNT {}",
             get_parser=float,
             vals=vals.Ints(1, 1048576),
```

Comment:
I've been taking a different tactic for validators, the trouble is if I want to say:
`awg.burst_count(1e5)` it will raise an error because `1e5` is a `float` rather than an `int`.

So instead I use a the `Numbers` validator for a given range, then have `set_parser=int`.

What do you think?

Reply:
I like it, I'll also change get_parser to int

Reply:
Changed